src/k8s-engine/wsl.ts

```diff
@@ -272,6 +277,9 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
   /** The current container engine; changing this requires a full restart. */
   #currentContainerEngine = ContainerEngine.NONE;
 
+  /** True if start() was called with k3s enabled, false if it wasn't. */
+  #enabledK3s = true;
```

Comment:
No, for the same reason as `this.#currentContainerEngine` and `this.cfg.containerEngine` -- when you change the setting in the UI, the config value is changed, and then `stop()` is run, and `stop` needs to know the current value on the object, not the desired value on the config.

Reply:
I thought that was no longer the case (because we now pass the config in via `stop`)? Or are we still managing to dynamically modify the object from far away?

Reply:
`config` is updated in real time. We don't modify the object's `#enabledK3s` remotely -- there's no writer on that field. But it tracks how the current running instance was created.

It's exactly the same as the case for `#currentContainerEngine` . I didn't test it, writing code for `#enabledK3s` write away, but I don't see how it could work when you change the config and then run `stop` without it.